### PR TITLE
Host: main & audio thread checks

### PR DESCRIPTION
**Pull request by Trinitou**
- I was not sure how to correctly checking for [thread-safe] so I added those as comments for now
- also had to make both `threadCheckIsMainThread` and `threadCheckIsAudioThread` `const` which makes sense in general but also because otherwise I could not use the thread checking methods inside const member functions
  - I guess not too many people use the host helper classes currently so I think it's OK to change it

### CODE REVIEW

**Comment by abique** [comment]
There is no check to be done for thread safe.

File: include/clap/helpers/host.hxx [hunk 1]
```diff
@@ -104,6 +119,7 @@ namespace clap { namespace helpers {
    const void *Host<h, l>::clapGetExtension(const clap_host_t *host,
                                             const char *extension_id) noexcept {
       auto &self = from(host);
+      //self.ensureThreadSafe("clap_host.get_extension");
```
Comment (abique) [inline include/clap/helpers/host.hxx]: You can get rid of that.

File: include/clap/helpers/host.hxx [hunk 2]
```diff
@@ -341,9 +383,41 @@ namespace clap { namespace helpers {
...
 
+   /////////////////////
+   // Thread Checking //
+   /////////////////////
+   template <MisbehaviourHandler h, CheckingLevel l>
+   void Host<h, l>::ensureMainThread(const char *method) const noexcept {
+      if (l == CheckingLevel::None)
+         return;
+
+      if (!threadCheckIsMainThread())
```
Comment (abique) [inline include/clap/helpers/host.hxx]: is the test inverted?

File: include/clap/helpers/host.hxx [hunk 3]
```diff
@@ -341,9 +383,41 @@ namespace clap { namespace helpers {
...
+          << "() on wrong thread! It must be called on main thread!";
+      pluginMisbehaving(msg.str());
+   }
+
+   template <MisbehaviourHandler h, CheckingLevel l>
+   void Host<h, l>::ensureAudioThread(const char *method) const noexcept {
+      if (l == CheckingLevel::None)
+         return;
+
+      if (!threadCheckIsAudioThread())
```
Comment (abique) [inline include/clap/helpers/host.hxx]: Again, test seems inverted.

File: include/clap/helpers/host.hxx (hunk 2 above)
Comment (Trinitou) [reply include/clap/helpers/host.hxx]: Oops, you're right. Corrected

File: include/clap/helpers/host.hxx (hunk 3 above)
Comment (Trinitou) [reply include/clap/helpers/host.hxx]: Oops, corrected as well

File: include/clap/helpers/host.hxx (hunk 1 above)
Comment (Trinitou) [reply include/clap/helpers/host.hxx]: Removed those comments everywhere

**Comment by Trinitou** [comment]
For `params.request_flush` I added a separate check that it is not called on audio-thread

File: include/clap/helpers/host.hxx [hunk 4]
```diff
@@ -235,12 +254,14 @@ namespace clap { namespace helpers {
...
                                     clap_param_clear_flags flags) noexcept {
       auto &self = from(host);
+      self.ensureMainThread("params.clear");
       self.paramsClear(param_id, flags);
    }
 
    template <MisbehaviourHandler h, CheckingLevel l>
    void Host<h, l>::clapParamsRequestFlush(const clap_host_t *host) noexcept {
       auto &self = from(host);
+      self.ensureAudioThread("params.request_flush", false);
```
Comment (abique) [inline include/clap/helpers/host.hxx:264]: This check is wrong, request flush must not be called on the audio thread and the function is supposed to be thread-safe.

Comment (Trinitou) [reply include/clap/helpers/host.hxx:264]: I added an `expectedState` parameter to `ensureAudioThread` which i set to `false` here

Comment (abique) [reply include/clap/helpers/host.hxx:264]: Ahhhh, then sorry :)

**Comment by abique** [comment]
Thanks!

**Pull request merged**